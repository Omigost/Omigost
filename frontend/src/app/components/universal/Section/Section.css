@import "../../../css/Theme.less";

.Section {
  margin-bottom: 10vw;
}

.SectionInverted {
  margin-bottom: 10vw;
  padding: 5vw;
  background: linear-gradient(to left, #f12711, #f5af19);
}

.SectionRotated {
  transform: rotate(5deg);
  width: 150vw;
  position: relative;
  left: -30vw;
  padding-bottom: 10vw;

  background: url('http://plinml.mimuw.edu.pl/img/backgrounds/speakers.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position-x: 17vw;
}

.SectionRotated .SectionContent {
  transform: rotate(-5deg);
}

.SectionContent {
  position: relative;
  left: 30vw;
}

@media only screen and (max-width: 425px) {
  .SectionRotated {
    min-height: 35vh;
  }
}
