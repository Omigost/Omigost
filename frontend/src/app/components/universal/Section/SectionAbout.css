@import "../../../css/Theme.less";

.SectionAbout {
  font-family: @PrimaryFont;
}

.AboutText {
  width: 80vw;
}