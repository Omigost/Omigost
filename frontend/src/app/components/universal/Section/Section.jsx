import React from 'react'
import './Section.css'
import ReactDOM from 'react-dom';

import { withRouter } from 'react-router'

import SectionHeader from 'components/SectionHeader/SectionHeader.jsx'
import SectionLeft from 'components/SectionLeft/SectionLeft.jsx'
import SectionRight from 'components/SectionRight/SectionRight.jsx'


class Section extends React.Component {
  constructor(props) {
      super(props);
      this.pageLocationParsed = false;
  }

  componentDidMount() {
      this.pageLocationParsed = false;
  }

  render() {
    let {
      children,
      title,
      inverted,
      rotated,
      icon,
      centeredTitle,
      id,
      withoutMargin,
      autolink,
      empty,
      rightContent
    } = this.props;

    let className = 'Section';
    if(inverted) {
      className = 'SectionInverted';
    }

    if(rotated) {
      className += ' SectionRotated';
    }

    if(centeredTitle === null ||
      typeof centeredTitle === 'undefined') {

      centeredTitle = true;
    }

    const { location } = this.props;

    if(location && location.hash && id && `#${id}` === location.hash) {
       if(this.pageLocationParsed === location.key) {

       } else {
           const scrollToLocationElement = (retryNo) => {
               let node = null;

               console.log('--> Try to find using jquery: #'+id);
               node = $(`#${id}`)[0];

               if(!node) {
                   try {
                       console.log('--> Try to find using ReactDOM');
                       node = ReactDOM.findDOMNode(this);
                   } catch(e) {
                       node = null;
                   }
               }

               if (!node) {
                   console.log('--> Failed');
                   setTimeout(() => {
                       scrollToLocationElement(retryNo + 1);
                   }, 500);
                   return;
               }

               console.log('--> Success');

               if (node.scrollIntoView) {
                   node.scrollIntoView();
                   window.scrollBy(0, -80);
               } else {
                   console.log('--> Scroll');
                   const vh = (Math.max(document.documentElement.clientHeight, window.innerHeight || 0) || 0) / 100.0;
                   console.log("VH = "+vh);
                   window.scrollTo(0, node.offsetTop + vh * 90);
               }
           };
           setTimeout(scrollToLocationElement, 0);
           this.pageLocationParsed = location.key || Math.random();
       }
    } else {
        this.pageLocationParsed = null;
    }

    if(empty) return (
      <div
        id={id}
        styleName={className}
      >
        <div></div>
      </div>
    );

    return (
      <div
        id={id}
        styleName={className}
      >
        <SectionHeader
          title={title}
          icon={icon}
          centered={rotated || centeredTitle}
          sectionId={id}
          autolink={autolink}
          inverted={inverted}
        />
        <div styleName='SectionContent'>
          <SectionLeft
            withoutMargin={withoutMargin}
          >
            {children}
          </SectionLeft>
          {
              (rightContent)?(
                <SectionRight>
                    {rightContent}
                </SectionRight>
              ):(null)
          }
        </div>
      </div>
    )
  }
}

export default withRouter(Section)
