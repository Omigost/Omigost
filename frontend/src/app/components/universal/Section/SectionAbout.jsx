import React from 'react'
import './SectionAbout.css'
import withContent from '../../utils/withContent.js'


import Section from 'components/Section/Section.jsx'
import CustomIcon from 'components/CustomIcon/CustomIcon.jsx'

class SectionAbout extends React.Component {
  render() {
    return (
      <div styleName='SectionAbout'>
        <Section
          withoutMargin
          id='about'
          title='About the conference'
        >
          <p styleName='AboutText'>
              {this.props.content.about}
          </p>
        </Section>
      </div>
    )
  }
}

export default withContent(SectionAbout)