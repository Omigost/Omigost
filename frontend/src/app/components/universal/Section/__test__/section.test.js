import React from 'react';
import renderer from 'react-test-renderer';

import Section from '../Section.jsx';

test('Snapshot test of Section', () => {
    const component = renderer.create(
        <Section
            title='My Section!'
            id='mysectionid'
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});