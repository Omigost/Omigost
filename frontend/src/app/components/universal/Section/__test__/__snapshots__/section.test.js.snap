// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Section 1`] = `
<div
  id="mysectionid"
  styleName="Section"
>
  <div
    styleName="SectionHeader"
  >
    <div
      styleName="CenteredTitleWrapper"
    >
      <a
        href="#mysectionid"
        styleName="TitleAutolink"
      >
        #
      </a>
      My Section!
    </div>
  </div>
  <div
    styleName="SectionContent"
  >
    <div
      styleName="SectionLeft"
    />
  </div>
</div>
`;
